samples/tutorials/mnist/04_Reusable_and_Pre-build_Components_as_Pipeline.ipynb

```diff
@@ -0,0 +1,799 @@
+    "# distributed under the License is distributed on an \"AS IS\" BASIS,\n",
+    "# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n",
+    "# See the License for the specific language governing permissions and\n",
+    "# limitations under the License.\n",
+    "# =============================================================================="
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Reusable components and Pre-build components\n",
+    "\n",
+    "This tutorial describes the manual way of writing a full component program (in any language) and a component definition for it. Below is a summary of the steps involved in creating and using a component:\n",
+    "\n",
+    "- Write the program that contains your component’s logic. The program must use files and command-line arguments to pass data to and from the component.\n",
+    "- Containerize the program.\n",
+    "- Write a component specification in YAML format that describes the component for the Kubeflow Pipelines system.\n",
+    "- Use the Kubeflow Pipelines SDK to load your component, use it in a pipeline and run that pipeline.\n",
+    "\n",
+    "More over, we will combine our built components together with a pre-build components and a lightweight component to compose a pipeline with three steps\n",
+    "- Train a mnist model and export to GCS\n",
+    "- Deploy the exported tensorflow model on AI Platform\n",
+    "- Test the deployment by calling the end point\n",
+    "\n",
+    "**Note: Make sure that you have docker installed in the local environment if you want to build the image locally**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import kfp\n",
+    "import kfp.gcp as gcp\n",
+    "import kfp.dsl as dsl\n",
+    "import kfp.compiler as compiler\n",
+    "import kfp.components as comp\n",
+    "import datetime\n",
+    "\n",
+    "import kubernetes as k8s"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "tags": [
+     "parameter"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Required Parameters\n",
+    "PROJECT_ID='<ADD GCP PROJECT HERE>'\n",
+    "GCS_BUCKET='gs://<ADD STORAGE LOCATION HERE>'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "tags": [
+     "parameter"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Optional Parameters, but required for running outside kubeflow cluster\n",
+    "HOST = '<ADD HOST NAME TO TALK TO KUBEFLOW PIPELINE HERE>'\n",
+    "CLIENT_ID = '<ADD OAuth CLIENT ID USED BY IAP HERE>'\n",
+    "OTHER_CLIENT_ID = '<ADD OAuth CLIENT ID USED TO OBTAIN AUTH CODES HERE>'\n",
+    "OTHER_CLIENT_SECRET = '<ADD OAuth CLIENT SECRET USED TO OBTAIN AUTH CODES HERE>'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create client\n",
```

Comment:
Please apply the same comments as previous notebooks.

Reply:
This has been changed to align with previous notebooks.